jwql/jwql_monitors/monitor_filesystem.py

```diff
@@ -133,15 +137,82 @@ def get_global_filesystem_stats(general_results_dict):
         A dictionary for the ``filesystem_general`` database table
     """
 
-    command = "df {}".format(FILESYSTEM)
+    command = "df -k {}".format(FILESYSTEM)
     command += " | awk '{print $3, $4}' | tail -n 1"
     stats = subprocess.check_output(command, shell=True).split()
-    general_results_dict['used'] = int(stats[0]) / (2**40)
-    general_results_dict['available'] = int(stats[1]) / (2**40)
+    general_results_dict['used'] = int(stats[0]) / (1024**3)
+    general_results_dict['available'] = int(stats[1]) / (1024**3)
 
     return general_results_dict
 
 
+def get_area_stats(central_storage_dict):
+    """Gathers ``used`` and ``available`` ``df``-style stats on the
+    selected area.
+
+    Parameters
+    ----------
+    central_storage_dict : dict
+        A dictionary for the ``central_storage`` database table
+
+    Returns
+    -------
+    central_storage_dict : dict
+        A dictionary for the ``central_storage`` database table
+    """
+    logging.info('Searching central storage system...')
+
+    arealist = ['logs', 'outputs', 'test', 'preview_images', 'thumbnails', 'all']
+    counteddirs = []
+
+    sum = 0  # to be used to count 'all'
+    for area in arealist:
+
+        used = 0
+        # initialize area in dictionary
+        if area not in central_storage_dict:
+            central_storage_dict[area] = {}
+
+        if area == 'all':
+            fullpath = CENTRAL
+        else:
+            fullpath = os.path.join(CENTRAL, area)
+
+        print(area, fullpath)
+        # record current area as being counted
+        counteddirs.append(fullpath)
+        print(counteddirs)
+
+        # to get df stats, use -k to get 1024 byte blocks
+        command = "df -k {}".format(fullpath)
+        command += " | awk '{print $2, $3, $4}' | tail -n 1"
+        stats = subprocess.check_output(command, shell=True).split()
+        # to put in TB, have to multiply values by 1024 to get in bytes, then
+        # divide by 1024 ^ 4 to put in TB
+        total = int(stats[0]) / (1024 ** 3)
+        free = int(stats[2]) / (1024 ** 3)
+        central_storage_dict[area]['size'] = total
+        central_storage_dict[area]['available'] = free
+
+        # do an os.walk on each directory to count up used space
+
+        for dirpath, _, files in os.walk(fullpath):
```

Comment:
@cracraft As you elude to in your comments, this is definitely the part that is taking the most time.  I wonder if it would be faster to do a `du -s` command here and use `subprocess.check_output` to get the results (akin to what you do for `df`), instead of measuring each file individually.  A crude test on my end shows that running `du` on the `preview_images` directory takes ~28 minutes to complete.  Considering `thumbnails` would be about the same size of directory, and the other directories are much smaller, I would reckon that the whole process would take roughly an hour.

Does that seem faster than what is currently implemented?

Reply:
I did a test of the code as it stands, and the file walking method took about half an hour to get through logs, outputs, test, preview_images, and thumbnails. It then took another hour to calculate for jwql, the upper level directory, that was also double-counting the earlier mentioned directories. I'd like to figure out how to avoid the double-counting that it's doing, and I think it would speed up quite a bit.